lib/docker/image.rb

```diff
@@ -28,7 +30,7 @@ def push(creds = nil, options = {}, &block)
     repo, tag = Docker::Util.parse_repo_tag(repo_tag)
     raise ArgumentError, "Image does not have a name to push." if repo.nil?
 
-    body = ""
+    body = +""
```

Comment:
Hm. Should we swap these to something like `StringIO`? Or just an `Array` with a `#join`?

It's been a minute since I've looked at this, and an intentional mutable string seems bad.

Reply:
> an intentional mutable string seems bad.

Do you mean a mutable string would be bad in regards to memory use, or readability, or to how well people will understand the code?

From my perspective it’s _easiest_ to continue using a string (but mark it as mutable) because `StringIO` has a different API (call `#string` to read out its String value), and as you mention with `Array` we’d have to call `#join`. But if another option is _better_, of course I’m up for changing the implementation.

Reply:
I was thinking in terms of reallocations. I don't know how string concat works under the hood enough, but I'd assume that `#join` or `#string` would perform better there.

I think only method internals would need to change, since `<<` works for both `IO` and `String`, so I think backwards compatibility should also be fine?

If it seems worse to change how it's working, I'm ok merging as is.

Reply:
Without having benchmarked it, I would _think_ that object allocations are comparable for all three approaches. Mutable strings shouldn't be worse than how they have worked up until this point, right?

Also, I don't think docker-api is going to be in a performance hotspot. I personally use it to automatically start containers before running my tests, so a handful of extra allocations would not bother me. But there may be other use cases that I am not aware of.

Reply:
Agreed, that is fair that it's not likely to be a performance concern itself. Ok, will merge as is. Thanks!